custom_components/google_home/sensor.py

```diff
@@ -146,68 +142,34 @@ def device_state_attributes(self):
     def _get_alarms_data(self) -> List[Dict[Any, Any]]:
         """Update alarms data extracting it from coordinator"""
         device = self.get_device()
-        return self.as_dict(device.get_sorted_alarms())
+        return self.as_dict(device.get_sorted_alarms()) if device else []
```

Comment:
We return `as_dict` but if not a list an empty?

Reply:
Oh, maybe we should have better named it `as_list_of_dict` 😄 
![image](https://user-images.githubusercontent.com/10655107/112525697-6e9d8480-8da1-11eb-859e-fe23f5c54c3b.png)
We basically feed it either a list or empty list in this line and it returns a list of dictionaries or an empty list

Reply:
Yeah, `as_dict` actually returns list of dicts 🙂

Reply:
@leikoilja I still think this function is way too hacky and I'm going to remove it 🙂

Reply:
haha 😆

Reply:
haha, @KapJI, as long as we have a working way to return data to sensors since they require data as dict